eng/testing/AppleRunnerTemplate.sh

```diff
@@ -38,15 +35,25 @@ while true; do
         sleep 5
     fi
 done
+# Restart the simulator to make sure it is tied to the right user session
+xcode_version=11.4
+xcode_path="/Applications/Xcode${xcode_version/./}.app"
+simulator_app="$xcode_path/Contents/Developer/Applications/Simulator.app"
+pid=`ps aux | grep "$simulator_app" | grep -v grep | tr -s ' ' | cut -d ' ' -f 2`
+if [ ! -z "$pid" ]; then
+    sudo kill "$pid"
+fi
+open -a "$simulator_app"
+export XHARNESS_OUT="$EXECUTION_DIR/xharness-output"
 
-XHARNESS_OUT="$EXECUTION_DIR/xharness-output"
-
-dotnet xharness ios test --app="$APP_BUNDLE" \
-    --targets=$TARGET \
+dotnet xharness ios test \
```

Comment:
Here I would also recommend specifying `--timeout` and `--launch-timeout`.
The former is for the overall run, the latter for how long it takes to boot the simulator, install the app and start it.

The failures you are seeing (mlaunch fails to find a simulator) have a retry logic and can take time from what I've seen so before we fix it, it could help increasing the timeouts (say 15 minutes for the launch timeout and 30-40 for the overall timeout)